File: library/src/main/java/com/ypk/library/utils/DisplayUtil.java
package com.ypk.library.utils;

import android.content.Context;
import android.util.TypedValue;

/**
 * @author ypk
 * 创建日期：2021/9/15  14:48
 * 描述：
 */
public class DisplayUtil {

    public static int dp2px(Context context, Float dpValue) {
        if (context == null) return (int) (dpValue * 1.5f + 0.5f);
        float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

    /**
     * sp转px
     */
    public static int sp2px(Context context, Float spVal) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, spVal, context.getResources().getDisplayMetrics());
    }

    /**
     * px转sp
     */
    public static float px2sp(Context context, Float pxVal) {
        return pxVal / context.getResources().getDisplayMetrics().scaledDensity;
    }
}

File: library/src/main/java/com/ypk/library/utils/DisplayUtil.kt
package com.ypk.library.utils

import android.content.Context
import android.util.TypedValue

/**
 * Created by 陈岗不姓陈 on 2017/10/30.
 *
 *
 */
object DisplayUtil {
    @JvmStatic
    fun dp2px(context: Context?, dpValue: Float): Int {
        if (context == null) return (dpValue * 1.5f + 0.5f).toInt()
        val scale = context.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }

    /**
     * sp转px
     */
    @JvmStatic
    fun sp2px(context: Context, spVal: Float): Int {
        return TypedValue.applyDimension(
            TypedValue.COMPLEX_UNIT_SP,
            spVal, context.resources.displayMetrics
        ).toInt()
    }

    /**
     * px转sp
     */
    fun px2sp(context: Context, pxVal: Float): Float {
        return pxVal / context.resources.displayMetrics.scaledDensity
    }
}